types/sharedb/lib/sharedb.d.ts

```diff
@@ -142,19 +128,46 @@ export class Doc<T = any> extends EventEmitter {
     flush(): void;
 }
 
-export type QueryEvent = 'ready' | 'error' | 'changed' | 'insert' | 'move' | 'remove' | 'extra';
-export class Query extends EventEmitter {
+export interface DocEventMap<T> {
+    'load': () => any;
+    'no write pending': () => any;
+    'nothing pending': () => any;
+    'create': (source: any) => any;
+    'op': (ops: any[], source: any) => any;
```

Comment:
If we wanted to be a bit more clear, we could declare this as `ops: [any]`, since there is always exactly one item:
https://github.com/share/sharedb/blob/e29936ca2dffd187b3fcfa3ca1b7e83becb6a2dc/lib/client/doc.js#L614-L625